persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -661,7 +681,11 @@ sqlForeign fdef = T.concat $
     , " FOREIGN KEY("
     , T.intercalate "," $ map (escapeF . snd. fst) $ foreignFields fdef
     , ") REFERENCES "
-    , escapeE $ foreignRefTableDBName fdef
+    , -- It's a syntax error in SQLite to use a dot-qualified table name.
+      -- In general, it's not possible for SQLite to maintain foreign key
+      -- constraints across databases (which Persistent calls "schemas").
+      -- So we omit the schema here.
```

Comment:
Should we instead throw an exception here if there is a cross-schema foreign key reference?

Reply:
If we don't, one of two things will occur.

1. The table holding the fkey reference shares a schema with a table sharing the same name as the intended target. Then we will create a fkey reference to the _wrong_ table.
2. There is no such table sharing the schema with the origin table. In this case, `sqlite` will emit an error saying that there is no table with the name we're looking for in the origin table's schema.

As I write this out, case (1) seems to tell me that we _should_ throw an error if there is a cross-schema foreign key reference being attempted.

Reply:
How does the `persistent` team feel about using `error`? Can I use that to handle this case, or should I lift this function to a monad that can throw synchronous exceptions?